### PR TITLE
Clean GH workflow

### PR DESCRIPTION
**Pull request by Jasopaum**
Not sure it's the cleanest nor if it even works. Maybe it would be better to have a plain script to do that?

### CODE REVIEW

File: .github/workflows/client.yml [hunk 1]
```diff
@@ -49,76 +49,37 @@ jobs:
...
-          cd client
-          ./scripts/generate_client_env.sh > .env.curie
-          docker build -t arkhn/pyrog-client:curie --build-arg ENV_FILE=.env.curie .
-
-      - name: Build RMS
+      - name: Build image
         env:
-          IP: ${{ secrets.IP_VM_REIMS }}
-          STAGE: rms
+          IP: ${{ secrets[matrix.environment.secret] }}
```
Comment (elsiehoffet-94) [inline .github/workflows/client.yml:70]: Isn't this "" for demo, staging and dev?

File: .github/workflows/client.yml [hunk 2]
```diff
@@ -49,76 +49,37 @@ jobs:
...
+          ]
     steps:
       - uses: actions/checkout@v1
 
-      - name: Build staging and demo
-        run: |
-          cd client/
-          yarn build:docker:staging
-          yarn build:docker:dev
-          yarn build:docker:demo
```
Comment (elsiehoffet-94) [inline .github/workflows/client.yml:60]: Shouldn't we keep those? or add IP_STAGING and IP_DEMO in the strategy?

Comment (Jasopaum) [reply .github/workflows/client.yml:60]: Oops, indeed I need to add IP_staging, etc. in the GH secrets if I want to do that
Which makes me even less convinced that this is the best solution 😅

Comment (simonvadee) [reply .github/workflows/client.yml:60]: good catch @elsiehoffet-94. I don't think it's that bad to add the url of our environments in GH secrets BUT if you're not convinced there is an other solution:
- keep staging,demo,dev in the matrix and remove the "secret" field (or keep in empty)
- do not generate a .env.<target> file for environments which do not have the "secret" field in the matrix --> use the .env.[dev|staging|demo] that we push in the repo instead

File: .github/workflows/client.yml (hunk 1 above)
Comment (simonvadee) [reply .github/workflows/client.yml:70]: yup, this wont work as is

**Comment by simonvadee** [comment]
I think it's pretty clever/clear to use the `matrix` feature of GH actions + a script using an env var (IP_ENV). I'm not sure if a plain script would make it clearer.

**Pull request merged**